### PR TITLE
E2E test where unordered channel is used with ICA

### PR DESCRIPTION
**Pull request by chatton**
## Description

~Review after: #5608~
closes: #5525

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/interchain_accounts/base_test.go [hunk 1]
```diff
@@ -26,6 +26,13 @@ import (
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
 )
 
+// orderMapping is a mapping from channel ordering to the string representation of the ordering.
+// the representation can be different depending on the relayer implmentation.
```
Comment (DimitrisJim) [inline e2e/tests/interchain_accounts/base_test.go:30]: bummer :sweat_smile:

**Review by DimitrisJim — APPROVED** [review]
short and sweet, lgtm!

File: e2e/tests/interchain_accounts/base_test.go (hunk 1 above)
Comment (charleenfei) [inline e2e/tests/interchain_accounts/base_test.go:30]:
```suggestion
// the representation can be different depending on the relayer implementation.
```

**Review by charleenfei — APPROVED** [review]
nit typo but can be fixed in any upcoming PR as well! thanks @chatton

**Review by crodriguezvega — APPROVED** [review]
Nice!

File: e2e/tests/interchain_accounts/base_test.go (hunk 1 above)
Comment (damiannolan) [reply e2e/tests/interchain_accounts/base_test.go:30]: this must be something to do with how the enum value is serialized with JSON or something maybe? 

So `channeltypes.UNORDERED` evaluates to `ORDER_UNORDERED` in normal go code but here one of the relayers returns it as `Unordered`? is it hermes?

**Review by damiannolan — APPROVED** [review]
Nice one!

File: e2e/tests/interchain_accounts/base_test.go (hunk 1 above)
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/base_test.go:30]:
```suggestion
// the representation can be different depending on the relayer implementation.
```

**Comment by crodriguezvega** [comment]
There are plenty of approvals, so I will merge the PR. There's an unrelated failing test, but I am looking into fixing it now.

**Pull request merged**